### DISCUSSION TITLE
Solo2 with Bitwarden extension crashes Safari

### DISCUSSION THREAD

**Discussion by phlbnks** [solokeys/solo2#106] [General]
Not sure which end the problem is coming from, but reporting here in case it's useful / anyone has experienced and can help.

Using Solo2 (non-hacker) with firmware 1:20200101.9 and Bitwarden 1.32/1.32.1 on MacOS 12.3:
- when attempting to authenticate Safari opens a "window" prompting insertion and activation of key
- key activates on touch (light changes)
- Safari doesn't complete authentication and locks-up, cancel button doesn't work, browser has to be force quit

Device restart and Bitwarden re-install no help.  

Any help appreciated, or if there's any more info I can give please let me know.

**Comment by nickray** [comment]
I just tested this with an M1 Mac mini via latest browser extension (Bitwarden 1.32.1) on Safari and "it works" (albeit with the preview firmware https://gist.github.com/nickray/7de5d40d5d93b7a1b6d0983062c9339e - this fixes some USB issues but breaks NFC, so don't update blindly).

<img width="440" alt="Screenshot 2022-03-26 at 00 04 05" src="https://user-images.githubusercontent.com/1092423/160212940-84c4ecfa-0404-4f42-a460-8b742071821a.png">

**Comment by phlbnks** [comment]
@nickray Thanks for your input. 

It's an improvement but still not properly working (also commented on your firmware gist):
Using Solo2 (non-hacker) with firmware 1:20220317.0 and Bitwarden 1.32.1 on MacOS 12.3 (Intel):
- when attempting to authenticate Safari opens a "window" prompting insertion and activation of key
- key activates on touch (light changes resting red, flash green as touched then to solid blue, blue flash, slow blue pulse, solid blue)
- Safari doesn't complete authentication but *doesn't* lock-up. I can Cancel out of the process and continue.

I have also reported this to Bitwarden fwiw. It's possible this firmware has fixed the lock-up problem I saw and now there's a corresponding issue in Bitwarden (or MacOS). Authentication succeeds in Chrome and fails (differently and before the key is inserted) in Firefox Developer and Standard editions.

**Reply by hoos97** [reply]
Is this only with Bitwarden?  Or are you able to complete registration on other sites through MacOS?

**Reply by phlbnks** [reply]
@hoos97 I also see this problem with Gitlab with current firmware so I presume it's Safari/SoloKey specific.

With the dev firmware Gitlab works, but Bitwarden still doesn't (browser doesn't lock-up though, so progress). 

There is definitely something weird with Bitwarden, as in Firefox I get an error even before inserting the key - but that seems in addition and doesn't explain all I'm seeing.

Chrome & Brave work on both sites using both Firmwares


Anything else you want me to test let me know

**Comment by ssgelm** [comment]
I'm having the same issue with every attempt to use my solo2 on safari 12.3 (m1). Not too interested in trying this beta firmware given the known issues, but if you believe those are resolved I'd be happy to test a new version.

**Comment by Grishkaone** [comment]
Hello !

I'm having the exact same issue with somme website : a window appear with options. I select the security key, but when I touch the solo2, nothing happen. The cancer button doesn't work and I must force quit Safari. Exemple site : Fastmail.

But I'm not sure it's a Safari issue. If I try on Firefox, there is an error "Sorry, something went wrong". But It's working on Windows.

On Firefox :
![Capture d’écran 2022-04-23 à 10 42 39](https://user-images.githubusercontent.com/46840243/164887412-a81d7a10-7da3-4ea8-937c-8cf1e1a76f9a.png)

On Safari : 

https://user-images.githubusercontent.com/46840243/164887457-42a4569f-bec9-44e6-b37d-a6f8a18e1e80.mov



- macOS 12.3.1
- Safari 15.4

**Comment by mathiasrabe** [comment]
I think there are several issues with 1:20200101.9 and MacOS. I can't log in into Github with my Solokey and Safari for example.

**Reply by mathiasrabe** [reply]
With version 1:20220317.0 and deleting the old key entry in github and adding it again, it works now. I did not test NFC yet.
